File: src/main/java/shopping/product/domain/ProductRepository.java
package shopping.product.domain;


import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.Optional;

public interface ProductRepository {

    Product save(Product product);

    Optional<Product> findById(Long id);

    void deleteById(Long id);

    Page<Product> findAll(Pageable pageable);
}

File: src/main/kotlin/shopping/domain/product/ProductRepository.kt
package shopping.domain.product

import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable

interface ProductRepository {
    fun findAll(pageable: Pageable): Page<Product>

    fun findById(id: Long): Product?

    fun save(product: Product): Product

    fun deleteById(id: Long)

    fun existsById(id: Long): Boolean
}